packages/framework/tests/application-server.spec.ts

```diff
@@ -90,3 +91,29 @@ test('basic controller', async () => {
 
     await applicationServer.close();
 });
+
+describe('http worker', () => {
+    test('needed for publicDir', async () => {
+        const testing = createTestingApp({
+            controllers: [],
+            imports: [KernelModule.configure({ publicDir: 'public' })]
+        });
+
+        await testing.startServer();
+        const applicationServer = testing.app.get(ApplicationServer);
+        expect(applicationServer.getWorker()).toBeInstanceOf(WebWorker);
+        await testing.stopServer();
+    })
+
+    test('not needed without controllers or publicDir', async () => {
```

Comment:
I think you missed a few test cases like testing for presence of RPC controllers.

Basically the 3 conditions present in https://github.com/deepkit/deepkit-framework/pull/118/files#diff-5dbe5852ace0e15a9424e45cfb23317e0bf65c0067d9406b506bb38d8ca92979R146 should be covered